lib/features/splash/presentation/splash_page.dart

```diff
@@ -25,10 +27,21 @@ class _SplashPageState extends State<SplashPage>
       parent: _animationController,
       curve: Curves.easeInOut,
     );
-    Future.delayed(const Duration(seconds: 2), () {
-      Navigator.of(context).pushReplacement(
-        MaterialPageRoute(builder: (context) => const SignInPage()),
-      );
+    Future.delayed(const Duration(seconds: 2), () async {
+      SharedPreferences prefs = await SharedPreferences.getInstance();
+      String? token = prefs.getString('authToken');
+      if (token != null && token.isNotEmpty) {
+        // ignore: use_build_context_synchronously
+        Navigator.of(context).pushReplacement(
+          MaterialPageRoute(
+              builder: (context) => const HomePage(initialIndex: 0)),
+        );
+      } else {
+        // ignore: use_build_context_synchronously
+        Navigator.of(context).pushReplacement(
+          MaterialPageRoute(builder: (context) => const SignInPage()),
+        );
+      }
```

Comment:
Or, you can improve readability just a little bit like...

```dart
final hasSignedIn = token != null && token.isNotEmpty;
final nextPage = hasSignedIn? const HomePage(initialIndex: 0) : const SignInPage();

  Navigator.of(context).pushReplacement(
    MaterialPageRoute(builder: (context) => nextPage),
  );
```

Reply:
really cool. thank you.